src/Microsoft.IdentityModel.JsonWebTokens/Json/JsonClaimSet.cs

```diff
@@ -0,0 +1,414 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#if !NET45
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Security.Claims;
+using System.Text.Json;
+using Microsoft.IdentityModel.Json.Linq;
+using Microsoft.IdentityModel.Logging;
+using Microsoft.IdentityModel.Tokens;
+
+namespace Microsoft.IdentityModel.JsonWebTokens
+{
+    /// <summary>
+    /// This class provides an abstraction over the json parser for net461+ using System.Text.Json.
+    /// <see cref="JsonWebToken"/> will delegate to this class to get values.
+    /// </summary>
+    internal class JsonClaimSet
+    {
+        private IList<Claim> _claims;
+        private static Type _typeofDateTime = typeof(DateTime);
```

Comment:
>_typeofDateTime = typeof(DateTime) [](http://example.com/codeflow?start=28&length=34)

Is this meant to improve performance? Since typeof(DateTIme) is resolved at compile-time, it should not have extra cost at runtime. Like other typeof(), it should be in the module MetadataToken.